lang/src/org/partiql/lang/planner/transforms/LogicalResolvedToPhysicalVisitorTransform.kt

```diff
@@ -0,0 +1,94 @@
+package org.partiql.lang.planner.transforms
+
+import org.partiql.lang.domains.PartiqlLogicalResolved
+import org.partiql.lang.domains.PartiqlLogicalResolvedToPartiqlPhysicalVisitorTransform
+import org.partiql.lang.domains.PartiqlPhysical
+
+/**
+ * Transforms an instance of [PartiqlLogicalResolved.Statement] to [PartiqlPhysical.Statement],
+ * specifying `(impl default)` for each relational operator.
+ */
+internal fun PartiqlLogicalResolved.Plan.toPhysicalPlan() =
+    LogicalResolvedToPhysicalVisitorTransform().transformPlan(this)
+
+internal val DEFAULT_IMPL = PartiqlPhysical.build { impl("default") }
+
+internal class LogicalResolvedToPhysicalVisitorTransform : PartiqlLogicalResolvedToPartiqlPhysicalVisitorTransform() {
+
+    /** Copies [PartiqlLogicalResolved.Bexpr.Scan] to [PartiqlPhysical.Bexpr.Scan], adding the default impl. */
+    override fun transformBexprScan(node: PartiqlLogicalResolved.Bexpr.Scan): PartiqlPhysical.Bexpr {
```

Comment:
For this and others that apply: can we add `withDefaultImpl` suffix (or something similar) to the names. Also where in the APIs are we're receiving `impl` as a parameter. (sorry if it's already somewhere and I've missed it).

Reply:
I'm not sure I understand the question exactly... but... 1) it is expected that a customer-supplied pass will change these from `(impl default)` to something that is customer specific.  2) right now, the `impl` is ignored by `PhysicalBexprToThunkConverterImpl` (introduced in #592), but I expect to fix that in a future PR that has as yet not been produced.  This will probably be the same PR that introduces the API that allows customers add passes that rewrite physical plans.

Reply:
Can we change the class name to `LogicalResolvedToPhysicalDefaultVisitorTransform` to indicate that users should have their own `impl` and implement `PartiqlLogicalResolvedToPartiqlPhysicalVisitorTransform` by themselves?

Reply:
> I'm not sure I understand the question exactly... but... 1) it is expected that a customer-supplied pass will change these from (impl default) to something that is customer specific. 2) right now, the impl is ignored by PhysicalBexprToThunkConverterImpl (introduced in #592), but I expect to fix that in a future PR that has as yet not been produced. This will probably be the same PR that introduces the API that allows customers add passes that rewrite physical plans.

So will customers be able to use `default` for some of the operators and pass their own for others in a pass?

Reply:
> So will customers be able to use default for some of the operators and pass their own for others in a pass?

That is the intent, yes, however, the `default` implementation will not always be the most performant or scalable.  In such circumstances, it is expected that the customer will supply their own implementation.  For instance, the user may supply an implementation of `distinct` which stores "seen" values on disk while the `default` implementation would do this using heap.

Reply:
> Can we change the class name to LogicalResolvedToPhysicalDefaultVisitorTransform to indicate that users should have their own impl and implement PartiqlLogicalResolvedToPartiqlPhysicalVisitorTransform by themselves?

Going with `LogicalResolvedToDefaultPhysicalVisitorTransform` and `toDefaultPhysicalPlan()`